.github/workflows/hip.yml

```diff
@@ -28,7 +38,8 @@ jobs:
             -DAMReX_AMD_ARCH=gfx908                       \
             -DCMAKE_C_COMPILER=$(which clang)             \
             -DCMAKE_CXX_COMPILER=$(which hipcc)           \
-            -DCMAKE_Fortran_COMPILER=$(which gfortran)
+            -DCMAKE_Fortran_COMPILER=$(which gfortran)    \
+            -DCMAKE_CXX_STANDARD=17
```

Comment:
Since this will only test C++17 now with HIP: Do we bump our HIP requirement to C++17 builds?

Reply:
I made the change simply because the name has c++17 in it. `name: HIP ROCm GFortran@9.3 C++17 [tests]`

We don't require C++17.  So we could add another to test C++14.

Reply:
Or change the name.

Reply:
Note that we only support recent rocm releases and they all support C++17.

Reply:
Yep, let's just convert the test to C++14 (maybe no need for another for the sake of CI time?) until we officially drop C++14 for HIP? Just requiring C++17 already for HIP (but then we need to doc this) is fine for me, too. We do the same for DPC++ (there we have no choice).

Reply:
I looked at `hip.yml` again.  We have a gnu make based test there that uses c++14.  So both standards are covered and I think we should leave it as is.